File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merged_list = nums1 + nums2
        merged_list = sorted(merged_list)
        n = len(merged_list)
        if n & 1:
            return float(f"{merged_list[n // 2]:.5f}")
        else:
            new_no = (merged_list[n // 2] + merged_list[(n // 2) - 1]) / 2
            return float(f"{new_no:.5f}")

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int n1 = nums1.length, n2 = nums2.length;
        // first combine the array
        int[] arr = new int[n1 + n2];
        int i = 0, j = 0, k = 0;
        while(i < n1 && j < n2) {
            arr[k++] = nums1[i] <= nums2[j] ? nums1[i++] : nums2[j++];
        }
        while(i < n1) {
            arr[k++] = nums1[i++];
        }
        while(j < n2) {
            arr[k++] = nums2[j++];
        }

        int n3 = arr.length;
        if((n3 & 1) == 1) {
            return arr[n3 / 2];
        } else {
            return (arr[n3 / 2] + arr[(n3 / 2) - 1]) / 2.0;
        }
    }
}